File: 3096. Minimum Levels to Gain More Points.java
class Solution {
    public int minimumLevels(int[] possible) {
        int total = 0;
        for (int i = 0; i < possible.length; i++) {
            if (possible[i] == 0) possible[i] = -1;
            total += possible[i];
        }
        int sum = 0;
        for (int i = 0; i < possible.length-1; i++) {
            sum += possible[i];
            total -= possible[i];
            if (sum > total) return i+1;
        }
        return -1;
    }
}

File: 3096. Minimum Levels to Gain More Points.go
func minimumLevels(possible []int) int {
	total, sum := 0, 0
	for i := range possible {
		if possible[i] == 0 {
			possible[i] = -1
		}
		total += possible[i]
	}
	for i := range len(possible)-1 {
		sum += possible[i]
		total -= possible[i]
		if sum > total {
			return i + 1
		}
	}
	return -1
}